draft-ietf-quic-v2.md

```diff
@@ -208,12 +208,35 @@ are willing to suffer a round-trip penalty if they are incorrect.
 # Applicability
 
 This version of QUIC provides no change from QUIC version 1 relating to the
-capabilities available to applications. Therefore, all Application Layer
+capabilities available to applications. Therefore, most Application Layer
 Protocol Negotiation (ALPN) ({{?RFC7301}}) codepoints specified to operate over
-QUICv1 can also operate over this version of QUIC.
+QUICv1 can also operate over this version of QUIC. Applications SHOULD NOT
+require new ALPNs for QUICv2 unless there is a specific performance or
+interoperability concern.
 
 All QUIC extensions defined to work with version 1 also work with version 2.
 
+## The HTTP/3 Exception
+
+The Alt-Svc HTTP field {{?RFC7838}} allows an HTTP server to inform clients of
+which versions of HTTP it supports. However, in order to use this information, clients
+also need to select a transport protocol that can carry HTTP.
+If multiple QUIC versions were to use the same ALPN to refer to HTTP/3, a
+client might discover QUIC support via Alt-Svc on a TCP connection, and then
+attempt to use a QUIC version that the server does not support. Therefore, HTTP/3 is an
+exception to the principle that new QUIC versions do not require new ALPN
+codepoints.  
+
+Therefore, HTTP/3 servers using QUIC version 2 MUST accept the "h3q2-01" ALPN
+for QUICv2 connections. Clients SHOULD request this ALPN over QUICv2. Servers
+MAY accept the "h3" ALPN over version 2 to improve compatibility with QUICv2
+clients supporting unmodified applications, but MUST NOT advertise the "h3"
+ALPN via Alt-Svc unless they also support QUIC version 1.
```

Comment:
wrong line (sorry) but this MUST NOT advertise "h3" requirement seems hard to implement in practice. Plus the server advertising "h3" could be a totally different entity to the alternative. Consider, in a new sentence, something like

The "h3" ALPN identifier indicates support for HTTP/3 over QUIC version 1. It SHOULD only be used in Alt-Svc when the alternative is known to support QUIC version 1.